python/llm/README.md

```diff
@@ -1,140 +1,137 @@
+1. [CLI (command line interface) Tool](#cli-tool)
+2. [Hugging Face `transformer`-style API](#hugging-face-transformers-style-api)
+3. [LangChain API](#langchain-api)
+4. [`llama-cpp-python`-style API](#llama-cpp-python-style-api)
+
+#### CLI Tool
+Currently `bigdl-llm` CLI supports *LLaMA* (e.g., *vicuna*), *GPT-NeoX* (e.g., *redpajama*), *BLOOM* (e.g., *pheonix*) and *GPT2* (e.g., *starcoder*) model architecture; for other models, you may use the `transformer`-style or LangChain APIs.
+
+ - ##### Convert model
+ 
+    You may convert the downloaded model into native INT4 format using `llm-convert`.
+    
+   ```bash
+   #convert PyTorch (fp16 or fp32) model; 
+   #llama/bloom/gptneox/starcoder model family is currently supported
+   lm-convert "/path/to/model/" --model-format pth --model-family "llama" --outfile "/path/to/output/"
+
+   #convert GPTQ-4bit model
+   #only llama model family is currently supported
+   llm-convert "/path/to/model/" --model-format gptq --model-family "llama" --outfile "/path/to/output/"
+   ```  
+  
+ - ##### Run model
+   
+   You may run the converted model using `llm-cli` (*built on top of `main.cpp` in [llama.cpp](https://github.com/ggerganov/llama.cpp)*)
+
+   ```bash
+   #help
+   #llama/bloom/gptneox/starcoder model family is currently supported
+   llm-cli -x llama -h
+
+   #text completion
+   #llama/bloom/gptneox/starcoder model family is currently supported
+   llm-cli -t 16 -x llama -m "/path/to/output/model.bin" -p 'Once upon a time,'
+   ```
+   
+#### Hugging Face `transformers`-style API
+You may run the models using `transformers`-style API in `bigdl-llm`
+
+- ##### Using native INT4 format
+
+   You may convert Hugging Face *Transformers* models into native INT4 format for maximum performance as follows. (Currently only *llama*/*bloom*/*gptneox*/*starcoder* model family is supported; for other models, you may use the [Hugging Face `transformers` format](hugging-face-transformers-style-api)).
+
+   ```python
+  #convert the model
+  from bigdl.llm import llm_convert
+  bigdl_llm_path = llm_convert(model='/path/to/model/',
+      outfile='/path/to/output/', outtype='int4', model_family="llama")
+
+  #load the converted model
+  from bigdl.llm.transformers import BigdlForCausalLM
+  llm = BigdlForCausalLM.from_pretrained("/path/to/output/model.bin",...)
+   
+  #run the converted  model
+  input_ids = llm.tokenize(prompt)
+  output_ids = llm.generate(input_ids, ...)
+  output = llm.batch_decode(output_ids)
+  ``` 
+
+  See the example here.
+
+- ##### Using Hugging Face `transformers` format
+
+  You may apply INT4 optimizations to any Hugging Face *Transformers* models as follows.
+
+  ```python
+  #load Hugging Face Transformers model with INT4 optimizations
+  from bigdl.llm.transformers import AutoModelForCausalLM
+  model = AutoModelForCausalLM.from_pretrained('/path/to/model/', load_in_4bit=True)
+
+  #run the optimized model
+  from transformers import LlamaTokenizer
```

Comment:
Maybe 
```python
from transformers import AutoTokenizer
```
is more appropriate.

(The reason why `LlamaTokenizer` is specifically used in the example is that the tokenizer class name in "decapoda-research/llama-7b-hf" is wrong.)

Reply:
actually there is another reason. `AutoTokenizer` is SUPER slow for llama tokenizer (~10min loading) while LlamaTokenizer is much better